kvdb/leveldb/leveldb.go

```diff
@@ -323,6 +324,18 @@ func (b *batch) Replay(w kvdb.Writer) error {
 	return b.b.Replay(&replayer{writer: w})
 }
 
+func (b *batch) DeleteRange(start, end []byte) error {
+	for it := b.db.NewIterator(util.BytesPrefix(nil), nil); it.Next(); {
+		key := it.Key()
+		if bytes.Compare(key, start) < 0 || bytes.Compare(key, end) >= 0 {
+			continue // skip keys that are not in the range
+		}
+		b.b.Delete(key)
+		b.size++
+	}
```

Comment:
The way it is here, the delete operation requires a full table scan. I think this can be restricted to only the relevant key range using this:
```suggestion
	for it := b.db.NewIterator(&util.Range{Start: start, Limit: end}, nil); it.Next(); {
		key := it.Key()
		b.b.Delete(key)
		b.size++
	}
```

Reply:
This is way nicer, thank you.